python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -0,0 +1,722 @@
+            max_shift: float maximum allowed shift to match patterns in
+                arcseconds.
+            max_rotation: float maximum allowed rotation between patterns in
+                degrees.
+            max_dist: float value specifying the maximum distance in
+                arcseconds allowed between candidate spokes in the source and
+                reference objects. Also sets that maximum distance in the
+                intermediate verify, pattern shift/rotation agreement, and
+                final verify steps.
+
+        Returns:
+            tuple (2D int array of matched pairs,
+                   float array of pair distances,
+                   int index of matched pattern)
+        """
+        return self.match(source_catalog=source_catalog, n_check=n_check,
+                          n_match=n_match, max_n_patterns=max_n_patterns,
+                          n_agree=1, max_shift=max_shift,
+                          max_rotation=max_rotation, max_dist=max_dist,
+                          min_matches=min_matches,
+                          pattern_skip_array=pattern_skip_array)
+
+    def match(self, source_catalog, n_check, n_match, n_agree,
+              max_n_patterns, max_shift, max_rotation, max_dist,
+              min_matches, pattern_skip_array=None):
+        """Function for matching a given source catalog into the loaded
+        reference catalog.
+        ----------------------------------------------------------------------
+        Args:
+            source_catalog: float array of spherical x,y,z coordinates and a
+                magnitude.
+            n_check: int value specifying the number of sources to attempt a
+                match on. Not all may be checked if n_match criteria is met
+                before hand. n_check should be greater than n_match by 1-3
+                objects.
+            n_match: int number of objects to use in constructing a pattern.
+            n_agree: int number of found patterns that must agree on their
+                shift and rotation before exiting.
+            max_n_patters:
+            max_shift: float maximum allowed shift to match patterns in
+                arcseconds.
+            max_rotation: float maximum allowed rotation between patterns in
+                degrees.
+            max_dist: float value specifying the maximum distance in
+                arcseconds allowed between candidate spokes in the source and
+                reference objects. Also sets that maximum distance in the
+                intermediate verify, pattern shift/rotation agreement, and
+                final verify steps.
+            pattern_skip_array: int array of patterns we would like to skip.
+                This could be due to the pattern being matched on a
+                pervious iteration that we now consider invalid.
+
+        Returns:
+            tuple (2D int array of matched pairs,
```

Comment:
I added a bit of text explaining the output. Once we finish will all of the more minor comments I'll start on the major ones if we need them like turning this output into a struct or creating a class for the tolerance.